dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/bundle/persister/EnrollmentPersister.java

```diff
@@ -117,6 +118,15 @@ protected boolean isNew(TrackerPreheat preheat, String uid) {
   @Override
   protected TrackerSideEffectDataBundle handleSideEffects(
       TrackerBundle bundle, Enrollment enrollment) {
+
+    SideEffectTrigger trigger = SideEffectTrigger.NONE;
+
+    if (isNew(bundle.getPreheat(), enrollment.getUid())) {
+      trigger = SideEffectTrigger.ENROLLMENT;
+    } else if (enrollment.isCompleted()) {
```

Comment:
This is not checking if the enrollment has been completed now, it is only checking that the status of the current enrollment is `COMPLETED`.
Every time a change will be made to an already completed enrollment, the side effect will be triggered

Reply:
Yes that is what we discussed in our last call. I am working on it.